src/esw/imu_driver.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+import rospy
+from sensor_msgs.msg import Imu, MagneticField, Temperature
+from geometry_msgs.msg import Quaternion, Vector3
+from std_msgs.msg import Header
+from mrover.msg import CalibrationStatus
+
+import serial
+from serial import SerialException, SerialTimeoutException
+
+
+def main():
+    # publishers for all types of IMU data, queue size is 1 to make sure we don't publish old data
+    imu_pub = rospy.Publisher("imu/data", Imu, queue_size=1)
+    mag_pub = rospy.Publisher("imu/magnetometer", MagneticField, queue_size=1)
+    temp_pub = rospy.Publisher("imu/temp", Temperature, queue_size=1)
+    calibration_pub = rospy.Publisher("imu/calibration", CalibrationStatus, queue_size=1)
+
+    # no rate used because rate is effectively set by Arduino serial publisher
+    rospy.init_node("imu_driver")
+
+    # read serial connection info and IMU frame from parameter server
+    port = rospy.get_param("imu_driver/port", "/dev/imu")
+    baud = rospy.get_param("imu_driver/baud", 115200)
+    imu_frame = rospy.get_param("imu_driver/frame_id", "imu_link")
+
+    # create serial connection with Arduino
+    ser = serial.Serial(port, baud)
```

Comment:
What happens if the port doesn't exist? Like if the user didn't set up /dev/imu or doesn't have an imu plugged in. Does the code just crash here? Also do we want to use ros log or something roserror to let the user get more info on why it crashed

Reply:
guthrie why did you wait until now to brind this up

Reply:
yeah it will crash and I think pyserial will throw an informative exception, I suppose you could roserror which would be more informative but rn its not terrible

Reply:
Kk sounds good